### PR TITLE
[Help wanted] Add kirigami2

### PR DESCRIPTION
**Pull request by ruilvo**
Specify library name and version:  **kirigami2/5.80.0**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

It's still not working, and it depends on #5010.

### CODE REVIEW

File: recipes/kirigami2/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    description = "Build beautiful, convergent apps that run on phones, TVs and everything in between."
+    license = "LGPLv2"
+    topics = ("conan", "gui", "qt")
+    homepage = "https://develop.kde.org/frameworks/kirigami/"
+    url = "https://github.com/conan-io/conan-center-index"
+    settings = "os", "arch", "compiler", "build_type"
+    generators = "cmake", "cmake_find_package"
+    requires = ["qt/5.15.2", "extra-cmake-modules/5.80.0"]
+    exports_sources = "CMakeLists.txt"
+
```
Comment (ericLemanissier) [inline recipes/kirigami2/all/conanfile.py:17]:
```suggestion

    @property
    def _build_subfolder(self):
        return "build_subfolder"

```

File: recipes/kirigami2/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        return "source_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = glob.glob("kirigami2*")[0]
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_cmake(self):
+        cmake = CMake(self)
+        cmake.configure()
```
Comment (ericLemanissier) [inline recipes/kirigami2/all/conanfile.py:29]:
```suggestion
        cmake.configure(build_folder=self._build_subfolder)
```

**Comment by ruilvo** [comment]
I need some help with this. But first #5010. needs to be merged. Then we'll go from there, if you decide to help me!

**Comment by ruilvo** [comment]
I'll close this for now because it's blocked by #4529 and #5019

**Pull request closed without merging**